plugins/transformations/seq2array/srcCxx/Seq2ArrayTransformation.cxx

```diff
@@ -0,0 +1,481 @@
+        break;
+    }
+
+    return is_compatible;
+}
+
+Seq2ArrayTransformation::Seq2ArrayTransformation(
+        const rti::routing::TypeInfo &input_type_info,
+        const rti::routing::TypeInfo &output_type_info,
+        const rti::routing::PropertySet &properties)
+        : input_type_info_(input_type_info.dynamic_type()),
+          output_type_info_(output_type_info.dynamic_type())
+{
+    if (!are_type_compatible(input_type_info_, output_type_info_)) {
+        throw std::runtime_error("Error: input and ouput types are not compatible");
+    }
+    // properties are not used because there is no additional configuration for
+    // this transformation
+}
+
+/*
+ * @brief Substitutes sequences by arrays of any type.
+ *
+ */
+void Seq2ArrayTransformation::convert_sample(
+        DynamicData &input_sample,
+        DynamicData &output_sample)
+{
+    // convert all the elements within a struct, sequence, array or union.
+    for (uint32_t i = 1; i <= input_sample.member_count(); ++i) {
+        uint32_t member_to_process = i;
+        bool is_union_member = false;
+
+        // unions only have to check the value that the discriminator identifies
+        if (input_sample.type_kind() == TypeKind::UNION_TYPE) {
+            member_to_process = input_sample.discriminator_value();
+            is_union_member = true;
+        }
+
+        if (!input_sample.member_exists(member_to_process)) {
+            continue;
+        }
+
+        // convert a member of the main Dynamic Data
+        switch (input_sample.member_info(member_to_process).member_kind().underlying()) {
+        case TypeKind::SEQUENCE_TYPE:
+        case TypeKind::ARRAY_TYPE: {
+            auto input_loaned_member = input_sample.loan_value(member_to_process);
+            auto output_loaned_member = output_sample.loan_value(member_to_process);
+
+            // in case of input sequences, check that the size of output array
+            // is enough to store all the values the sequence has.
+
+            // Get the size of the ouput array
+            const ArrayType &array_type =
+                    static_cast<const ArrayType &>(output_loaned_member.get().type());
+
+            // Check that the size of the ouput array is enough
+            if (input_loaned_member.get().member_count() > array_type.total_element_count()) {
+                std::string error("Error: not enough space to copy output field <"
+                        + output_loaned_member.get().type().name()
+                        + "> (max size <"
+                        + std::to_string(array_type.total_element_count())
+                        + ">) into input field <"
+                        + input_loaned_member.get().type().name()
+                        + "> (actual size <"
+                        + std::to_string(input_loaned_member.get().member_count())
+                        + ">).");
+                throw std::runtime_error(error);
+            }
```

Comment:
Have you considered allowing the transformation to truncate the sequence value?

Reply:
yes, we can add in the future some configuration about the strategy of copying elements of sequences/arrays (truncate or error)